build_snp_matrix.py

```diff
@@ -1,5 +1,252 @@
+                         "pcMapped":float, "Outcome":"category", "flag":"category",
+                         "group":"category", "CSSTested":float, "matches":float,
+                         "mismatches":float, "noCoverage":float, "anomalous":float})
+    return df
+
+def remove_duplicates(df, parameter="pcMapped"):
+    """
+        Drops duplicated submissions from df
+    """
+    return df.drop(get_indexes_to_remove(df, parameter))
+
+def get_indexes_to_remove(df, parameter):
+    """
+        Loops through unique submisions in the summary_df and collects indexes
+        for duplicate submisions which should be excluded.
+
+        Parameters:
+            df (pandas dataframe object): a dataframe read from btb_wgs_samples.csv
+
+            parameter (string): a column name from df on which to decide which 
+            duplicate to keep, e.g. keep the submission with the largest pcMapped
+
+        Returns:
+            indexes (pandas index object): indexes to remove from dataframe
+    """
+    indexes = pd.Index([])
+    for submission_no in df.submission.unique():
+        parameter_max = df.loc[df["submission"]==submission_no][parameter].max()
+        if len(df.loc[(df["submission"]==submission_no) & (df[parameter] == parameter_max)]) > 1:
+            print(f"Submision {submission_no} is duplicated and has the same {parameter} value\n"
+                    f"Skipping submision {submission_no}" )
+            # if duplicate submissions share the maximum paramter value add the all 
+            # entries with submission_no to the list of indexes to remove
+            indexes = indexes.append(df.loc[df["submission"]==submission_no].index)
+        else:
+            # otherwise add all duplicates except for the one with the maximum 
+            # parameter value
+            indexes = indexes.append(df.loc[(df["submission"]==submission_no) & \
+                (df[parameter] != parameter_max)].index)
+    return indexes
+
+def filter_df(df, pcmap_threshold=(0,100), **kwargs):
+    """ 
+        Filters the sample summary dataframe which is based off 
+        btb_wgs_samples.csv according to a set of criteria. 
+
+        Parameters:
+            df (pandas dataframe object): a dataframe read from btb_wgs_samples.csv.
+
+            pcmap_threshold (tuple): min and max thresholds for pcMapped
+
+            **kwargs: 0 or more optional arguments. Names must match a 
+            column name in btb_wgs_samples.csv. If column is of type 
+            'categorical' or 'object', vales must be of type 'list' 
+            ispecifying a set of values to match against the argument 
+            name's column in btb_wgs_samples.csv. For example, 
+            'sample_name=["AFT-61-03769-21", "20-0620719"]' will include 
+            just these two samples. If column is of type 'int' or 'float',
+            values must be of type 'tuple' and of length 2, specifying a 
+            min and max value for that column. 
+
```

Comment:
I assume this is where it would be possible to generate a list for a specific clade (group column in the csv)?